[2023-06-15T14:53:47.179+0700] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: thethree_reqif_to_rst.load_config_then_build_json manual__2023-06-15T07:53:33+00:00 [queued]>
[2023-06-15T14:53:47.185+0700] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: thethree_reqif_to_rst.load_config_then_build_json manual__2023-06-15T07:53:33+00:00 [queued]>
[2023-06-15T14:53:47.185+0700] {taskinstance.py:1308} INFO - Starting attempt 1 of 0
[2023-06-15T14:53:47.197+0700] {taskinstance.py:1327} INFO - Executing <Task(PythonOperator): load_config_then_build_json> on 2023-06-15 07:53:33+00:00
[2023-06-15T14:53:47.200+0700] {standard_task_runner.py:57} INFO - Started process 32230 to run task
[2023-06-15T14:53:47.206+0700] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'thethree_reqif_to_rst', 'load_config_then_build_json', 'manual__2023-06-15T07:53:33+00:00', '--job-id', '47', '--raw', '--subdir', 'DAGS_FOLDER/mainflow.py', '--cfg-path', '/var/folders/v7/qsb_cvl92z92z6jf7g1jc0zr0000gn/T/tmpcol4hro5']
[2023-06-15T14:53:47.208+0700] {standard_task_runner.py:85} INFO - Job 47: Subtask load_config_then_build_json
[2023-06-15T14:53:47.253+0700] {task_command.py:410} INFO - Running <TaskInstance: thethree_reqif_to_rst.load_config_then_build_json manual__2023-06-15T07:53:33+00:00 [running]> on host macbook-pro.local
[2023-06-15T14:53:47.297+0700] {taskinstance.py:1545} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='The Three' AIRFLOW_CTX_DAG_ID='thethree_reqif_to_rst' AIRFLOW_CTX_TASK_ID='load_config_then_build_json' AIRFLOW_CTX_EXECUTION_DATE='2023-06-15T07:53:33+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-06-15T07:53:33+00:00'
[2023-06-15T14:53:47.319+0700] {mainflow.py:24} INFO - Loading config from /Users/lap14202/.airflow/data/config.yml
[2023-06-15T14:53:47.328+0700] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/airflow/operators/python.py", line 181, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/airflow/operators/python.py", line 198, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lap14202/.airflow/dags/mainflow.py", line 26, in load_config_then_build_json_task
    json_data = build_json(reqif, config)
                ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lap14202/.airflow/thethree/build_json.py", line 249, in build_json
    artifacts = get_artifacts(reqif, config['artifacts']['artifact'])
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lap14202/.airflow/thethree/build_json.py", line 233, in get_artifacts
    artifacts.append(get_artifact(reqif, spec_object, config))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lap14202/.airflow/thethree/build_json.py", line 212, in get_artifact
    attributes = get_artifact_attributes(spec_object, definition, config)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lap14202/.airflow/thethree/build_json.py", line 182, in get_artifact_attributes
    attr_value = find_enum_value(attr['VALUES']['ENUM-VALUE-REF'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lap14202/.airflow/thethree/build_json.py", line 150, in find_enum_value
    find_keys(load_reqif('../Requirements.reqif'), 'DATATYPE-DEFINITION-ENUMERATION'))[0]
              ^^^^^^^^^^
NameError: name 'load_reqif' is not defined
[2023-06-15T14:53:47.334+0700] {taskinstance.py:1345} INFO - Marking task as FAILED. dag_id=thethree_reqif_to_rst, task_id=load_config_then_build_json, execution_date=20230615T075333, start_date=20230615T075347, end_date=20230615T075347
[2023-06-15T14:53:47.342+0700] {standard_task_runner.py:104} ERROR - Failed to execute job 47 for task load_config_then_build_json (name 'load_reqif' is not defined; 32230)
[2023-06-15T14:53:47.388+0700] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2023-06-15T14:53:47.405+0700] {taskinstance.py:2651} INFO - 0 downstream tasks scheduled from follow-on schedule check
